server/issue_test.go

```diff
@@ -279,12 +269,214 @@ func TestRouteShareIssuePublicly(t *testing.T) {
 	}
 }
 
-func TestRouteAttachCommentToIssue(t *testing.T) {
-	api := &plugintest.API{}
+func TestShouldReceiveNotification(t *testing.T) {
+	cs := ConnectionSettings{}
+	cs.RolesForDMNotification = make(map[string]bool)
+	cs.RolesForDMNotification[assigneeRole] = true
+	cs.RolesForDMNotification[mentionRole] = true
+	for name, tt := range map[string]struct {
+		role         string
+		notification bool
+	}{
+		assigneeRole: {
+			role:         assigneeRole,
+			notification: true,
+		},
+		mentionRole: {
+			role:         mentionRole,
+			notification: true,
+		},
+		reporterRole: {
+			role:         reporterRole,
+			notification: false,
+		},
+		watchingRole: {
+			role:         watchingRole,
+			notification: false,
+		},
+	} {
+		t.Run(name, func(t *testing.T) {
+			val := cs.ShouldReceiveNotification(tt.role)
+			assert.Equal(t, tt.notification, val)
+		})
+	}
```

Comment:
For completeness, we should also test the backwards compability case

Reply:
@mickmister Can you please elaborate on what you mean by `test the backward compatibility case`?

As earlier, in order to get a DM notification, the user has to meet certain criteria like being assigned or mentioned. But now we added support for DM subscriptions for the watchers and reporters as well.

Reply:
@Nityanand13 I mean test the case where `settings.Notifications` is `true` or `false`, and the rest of the user notification settings are not defined yet, because that's going to be the case for every current installation of the plugin